### PR TITLE
Laura/test epoch network change

### PR DESCRIPTION
**Pull request by laura-makdah**
This contains code that:
• tests the reconfiguration flow from an end-to-end level over a network, but without any validator changes
• fixes a bug with network recreation 
• adds programatic move call to add a new validator to the network

### CODE REVIEW

File: crates/sui-core/src/authority/authority_store.rs [hunk 1]
```diff
@@ -1502,6 +1502,13 @@ impl<S: Eq + Serialize + for<'de> Deserialize<'de>> SuiDataStore<S> {
     pub fn get_latest_authenticated_epoch(&self) -> Option<AuthenticatedEpoch> {
         self.epochs.iter().skip_to_last().next().map(|(_, a)| a)
     }
+
+    pub fn get_sui_system_state_object_ref(&self) -> SuiResult<ObjectRef> {
+        let sui_system_object = self
+            .get_object(&SUI_SYSTEM_STATE_OBJECT_ID)?
+            .expect("Sui System State object must always exist");
```
Comment (lxfind) [inline crates/sui-core/src/authority/authority_store.rs:1509]: There should already be a `get_sui_system_state_object`?

File: crates/sui-adapter/src/adapter.rs [hunk 2]
```diff
@@ -371,8 +374,11 @@ pub fn generate_package_id(
     let package_id = ctx.fresh_id();
     for module in modules.iter() {
         let old_module_id = module.self_id();
+        let package_name = old_module_id.name().as_str();
+        let is_framework =
+            package_name == SUI_PACKAGE_NAME || package_name == MOVE_STDLIB_PACKAGE_NAME;
         let old_address = *old_module_id.address();
-        if old_address != AccountAddress::ZERO {
+        if !is_framework && old_address != AccountAddress::ZERO {
```
Comment (lxfind) [inline crates/sui-adapter/src/adapter.rs:381]: What's the reason behind this change?

File: crates/sui-core/src/epoch/reconfiguration.rs [hunk 3]
```diff
@@ -192,29 +192,26 @@ where
...
-                });
-            let address: Multiaddr = str_addr
-                .unwrap()
-                .parse()
-                .map_err(|e: multiaddr::Error| SuiError::GenericAuthorityError {
-                    error: e.to_string(),
-                })
-                .unwrap();
+                }
+            })?;
```
Comment (lxfind) [inline crates/sui-core/src/epoch/reconfiguration.rs:199]: Please create an issue and add a TODO here so we can get back to this: one invalid validator should not block the entire reconfiguration process. However what to do with the invalid validator in the new committee is an open question.

File: crates/sui-node/src/lib.rs [hunk 4]
```diff
@@ -52,6 +52,7 @@ pub struct SuiNode {
     _execute_driver_handle: Option<tokio::task::JoinHandle<()>>,
     _checkpoint_process_handle: Option<tokio::task::JoinHandle<()>>,
     state: Arc<AuthorityState>,
+    active: Option<Arc<ActiveAuthority<NetworkAuthorityClient>>>,
```
Comment (lxfind) [inline crates/sui-node/src/lib.rs:55]: When is this None?

File: crates/sui/tests/reconfiguration_tests.rs [hunk 5]
```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
```
Comment (lxfind) [inline crates/sui/tests/reconfiguration_tests.rs:1]: Let's rename this file to maybe reconfig_end_to_end_tests to distinguish from the other reconfiguration_tests

**Review by lxfind — COMMENTED** [review]
Thanks! Looks good overall. Added a few comments

File: crates/sui-core/src/epoch/reconfiguration.rs (hunk 3 above)
Comment (laura-makdah) [reply crates/sui-core/src/epoch/reconfiguration.rs:199]: Yeah, this is a good question. I think the best place to check this is in the move code. We ideally handle this here as well, but it would be more of a business / protocol decision what to do.

File: crates/sui-node/src/lib.rs (hunk 4 above)
Comment (laura-makdah) [reply crates/sui-node/src/lib.rs:55]: when we have gossip turned off

File: crates/sui-adapter/src/adapter.rs (hunk 2 above)
Comment (laura-makdah) [reply crates/sui-adapter/src/adapter.rs:381]: It was a refactor, but I've reverted it since it is not related or neccessary.

File: crates/sui-core/src/authority/authority_store.rs (hunk 1 above)
Comment (laura-makdah) [reply crates/sui-core/src/authority/authority_store.rs:1509]: We might need the object_ref for other programatic move calls, but I'll remove it for now.

**Review by lxfind — APPROVED** [review]
